streams/src/main/java/org/apache/kafka/streams/processor/internals/StreamThread.java

```diff
@@ -233,6 +233,9 @@ public void onPartitionsAssigned(Collection<TopicPartition> assignment) {
                     StreamThread.this.getName(), assignment);
 
                 setStateWhenNotInPendingShutdown(State.ASSIGNING_PARTITIONS);
+                // do this first as we may have suspended standby tasks that
+                // will become active
+                closeNonAssignedSuspendedStandbyTasks();
```

Comment:
Should we move closing non-assigned tasks into this method too? Would be a cleaner workflow -- close all not reassigned tasks (regular and stand-by), than reuse suspended tasks and create new tasks.